### PR TITLE
Add requests verify option in vault section

### PR DESCRIPTION
**Pull request by sakateka**
This fix allow pass requests 'verify' option from vault configs
If vault certificate signed with Intermediate CA,
and Intermedia CA sign by internal root CA,
requests will fail verifying vault certificate with error:
```
 _ssl.c:510: ... routines:SSL3_GET_SERVER_CERTIFICATE:certificate verify failed
```
'verify' option allow explicitly specify ca-bundle, or disable verifications.
http://docs.python-requests.org/en/master/user/advanced/#ssl-cert-verification

### What does this PR do?
add 'verify' option in vault config
### Previous Behavior
requests the verify option is remained at the default value
### New Behavior
requests the verify option is passed from vault config if specified or remain in default value
### Tests written?
No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

### CODE REVIEW

**Comment by sakateka** [comment]
I'm sorry, I wanted to make a pull request to the branch 2017.7.2, but inadvertently made in the develop. 
The base branch of this PR was 2017.7.2.
May be need to close this PR and make a new one in 2017.7.2?

**Comment by cachedout** [comment]
@sakateka We're adding functionality here, so `develop` is the correct branch.

File: salt/modules/vault.py [hunk 1]
```diff
@@ -32,6 +33,10 @@
     url
         Url to your Vault installation. Required.
 
+    verify
+        For details please see
+        http://docs.python-requests.org/en/master/user/advanced/#ssl-cert-verification
+
```
Comment (rallytime) [inline salt/modules/vault.py]: Can you add a `.. versionadded:: Oxygen` tag here?

File: salt/utils/vault.py [hunk 2]
```diff
@@ -124,6 +126,8 @@ def make_request(method, resource, profile=None, **args):
 
     connection = _get_vault_connection()
     token, vault_url = connection['token'], connection['url']
+    if "verify" not in args:
```
Comment (rallytime) [inline salt/utils/vault.py]: I know this is picky, but we prefer single quotes instead of double. :)

**Review by rallytime — COMMENTED** [review]
@sakateka Thank you for submitting this. I only had a couple of very small comments.

**Review by rallytime — APPROVED** [review]

**Pull request merged**